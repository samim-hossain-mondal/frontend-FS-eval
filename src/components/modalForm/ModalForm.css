.modal {
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: 0;
  transition: all 0.3s ease-in-out;
  pointer-events: none;
}

.modal.enter-done {
  opacity: 1;
  pointer-events: visible;
}

.modal.exit {
  opacity: 0;
}

.modal-content {
  width: 500px;
  background-color: #fff;
  transition: all 0.3s ease-in-out;
  transform: translateY(-200px);
}

.modal.enter-done .modal-content {
  transform: translateY(0);
}

.modal.exit .modal-content {
  transform: translateY(-200px);
}

.modal-header{
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 20px;
  font-size: larger;
}

.modal-header > h3 {
  margin: 0;
  font-size: 1.5rem;
  color: rgb(78, 78, 255);
}
.modal-footer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 10px;
}

.modal-footer > button {
  background-color: white;
  color:  rgb(78, 78, 255);
  border-color:rgb(175, 175, 254);
  border-radius: 5px;
  padding: 0.3rem 0.8rem;
  font-size: 1.2rem;
  cursor: pointer;
}

.modal-body{
  padding: 0;
}

.modal-title {
  margin: 0;
}
