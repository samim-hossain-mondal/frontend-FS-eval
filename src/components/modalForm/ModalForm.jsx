import React, { useEffect } from 'react';
import proptypes from 'prop-types';
import { CSSTransition } from 'react-transition-group';

export default function Modal(props){
  const closeOnEscapeKeyDown = e => {
    if ((e.charCode || e.keyCode) === 27) {
      props.onClose();
    }
  };

  useEffect(() => {
    document.body.addEventListener('keydown', closeOnEscapeKeyDown);
    return function cleanup() {
      document.body.removeEventListener('keydown', closeOnEscapeKeyDown);
    };
  }, []);

  return (
    <CSSTransition
      in={props.show}
      unmountOnExit
      timeout={{ enter: 0, exit: 300 }}
    >
      <div className="modal" onClick={props.onClose}>
        <div className="modal-content" onClick={e => e.stopPropagation()}>
          <div className="modal-header">
            <h4 className="modal-title">{props.title}</h4>
          </div>
          <div className="modal-body">{props.children}</div>
          <button onClick={props.onClose} className="button">
              Close
          </button>
        </div>
      </div>
    </CSSTransition>
  );
}

Modal.propTypes = {
  onClose: proptypes.func.isRequired,
  show: proptypes.bool,
  title: proptypes.string,
  children: proptypes.node,
  onSave: proptypes.func
};


