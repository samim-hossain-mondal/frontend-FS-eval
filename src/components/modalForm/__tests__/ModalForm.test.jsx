import React from 'react';
import { render } from '@testing-library/react';

import ModalForm from '../ModalForm';

describe('ModalForm', () => {
  it('should render the ModalForm component', () => {
    const { container } = render(<ModalForm />);
    expect(container).toMatchSnapshot();
  });
});