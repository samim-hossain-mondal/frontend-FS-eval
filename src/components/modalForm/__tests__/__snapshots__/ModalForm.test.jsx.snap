// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalForm should render the ModalForm component 1`] = `<div />`;
